presto-main/src/main/java/com/facebook/presto/connector/ConnectorAwareNodeManager.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector;
+
+import com.facebook.presto.metadata.InternalNodeManager;
+import com.facebook.presto.spi.Node;
+import com.facebook.presto.spi.NodeManager;
+import com.facebook.presto.spi.NodeState;
+
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+public class ConnectorAwareNodeManager
```

Comment:
It will have the connector ID later. If you have a better name for this commit I can use it here then rename later.